contracts/peripheral/TokenDrip.sol

```diff
@@ -0,0 +1,166 @@
+    uint256 public target;
+
+    /// @dev The rate per second of the drip
+    uint256 public rate;
+
+    /* -------------------------------------------------------------------
+       Events 
+    ------------------------------------------------------------------- */
+
+    event UpdateDrip(uint256 lastUpdated, uint256 current, uint256 target, uint256 rate);
+    event Drip(uint256 amount);
+    event Cancel(uint256 lastUpdated, uint256 current, uint256 target, uint256 rate);
+
+    /* -------------------------------------------------------------------
+       Constructor 
+    ------------------------------------------------------------------- */
+
+    /**
+     * @param _token    The token address
+     * @param _to       The address to drip to
+     * @param _target   The target amount to drip in total
+     * @param _rate     The rate per second to drip tokens
+     */
+    constructor(
+        address _token,
+        address _to,
+        uint256 _target,
+        uint256 _rate
+    ) {
+        token = _token;
+        to = _to;
+
+        _updateDrip(block.timestamp, 0, _target, _rate);
+    }
+
+    /* -------------------------------------------------------------------
+       Core 
+    ------------------------------------------------------------------- */
+
+    /**
+     * @dev Drip token on the schedule
+     */
+    function drip() external {
+        require(target != 0, "!drip");
+
+        // get the amount of time past since the last drip and calculate how
+        // many tokens can be dripped based on the current rate
+        uint256 delta = block.timestamp.sub(lastUpdated);
+        uint256 amount = delta.mul(rate);
+        uint256 newCurrent = current.add(amount);
+
+        // If the newly calculated current amount is greater than the target
+        // then recalculate the amount as the difference between the target
+        // and the current value.
+        if (newCurrent > target) {
+            amount = target.sub(current);
+            // Update the new current value based on the amount
+            newCurrent = current.add(amount);
+        }
+
+        require(amount != 0, "!amount");
+
+        // Update storage
+        current = newCurrent;
+        lastUpdated = block.timestamp;
+
+        IERC20(token).transfer(to, amount);
+
+        emit Drip(amount);
+    }
+
+    /**
+     * @dev Update the drip configuration
+     * @param _lastUpdated  The last updated time
+     * @param _current      The current value that has been dripped
+     * @param _target       The target amount to drip in total
+     * @param _rate         The rate per second to drip tokens
+     */
+    function update(
+        uint256 _lastUpdated,
+        uint256 _current,
+        uint256 _target,
+        uint256 _rate
+    ) external onlyOwner {
+        _updateDrip(_lastUpdated, _current, _target, _rate);
+    }
+
+    /**
+     * @dev Cancel the current drip
+     */
+    function cancel() external onlyOwner {
```

Comment:
I am not that sure about this cancel fn: 
- Usually whenever any "stream" is cancel , whatever balance that could have been drip is dripped during the cancel process. Here it would be canceling only and not dripping what the target had "earned"

- Setting all to 0 will erase the record of what had beed already dripped, have you considered instead adjusting the target == current?  So later if needed it could be easily tracked what was the latest current.

Reply:
This is just allowing us to drip tokens at a certain rate. If the DAO is calling cancel its because they don't want to drip anymore tokens. This is fine.